juju_verify/verifiers/base.py

```diff
@@ -109,6 +110,27 @@ def unit_from_id(self, unit_id: str) -> Unit:
         raise VerificationError('Unit {} was not found in {} verifier.'
                                 ''.format(unit_id, self.NAME))
 
+    @staticmethod
+    def check_minimum_version(min_version: Version, units: List[Unit]) -> Result:
+        """Check minimum required version of juju agents on units."""
+        result = Result()
+        for unit in units:
+            juju_version = unit.safe_data.get('agent-status', {}).get('version', '')
+            try:
+                if Version(juju_version) < min_version:
+                    fail_msg = (f'Juju agent on unit {unit.entity_id} has lower than '
+                                f'minumum required version. {juju_version} < '
+                                f'{min_version}')
+                    result.add_partial_result(Severity.FAIL, fail_msg)
+            except InvalidVersion as exc:
+                raise CharmException(f'Failed to parse juju version from '
+                                     f'unit {unit.entity_id}.') from exc
+
+        if result.empty:
+            result.add_partial_result(Severity.OK, 'Minimum juju version check passed.')
```

Comment:
What do you say if we add `__bool__` to the Result class?
```python
def __bool__(self):
    return not self.empty
```
Than we can use this:
```python
return result or Result(Severity.OK, 'Minimum juju version check passed.')
```

This is not within the scope of this PR, just thinking of improvement.

Reply:
Sounds good.

Reply:
Ok, I will report a bug with this.